crates/iota-sdk/src/lib.rs

```diff
@@ -644,3 +636,87 @@ impl DataReader for ReadApi {
         Ok(self.get_reference_gas_price().await?)
     }
 }
+
+pub trait GetAllPages<O, C, F, E>: Sized + Fn(Option<C>) -> F
+where
+    O: Send,
+    C: Send,
+    F: futures::Future<Output = Result<Page<O, C>, E>> + Send,
+{
+    fn get_all_pages(self) -> impl futures::Future<Output = Result<Vec<O>, E>> {
```

Comment:
When you write the doc of this one, I would say to be careful, this could eat up a LOT of memory

Reply:
Eh it's kind of assumed. I added a doc which I think is fine